public/app/features/alerting/unified/components/rule-editor/ExpressionsEditor.tsx

```diff
@@ -45,7 +45,11 @@ export const ExpressionsEditor = ({
         const data = panelData[query.refId];
 
         const isAlertCondition = condition === query.refId;
-        const error = data ? errorFromPreviewData(data) : undefined;
+
+        const errorFromCondition = data && isAlertCondition ? errorFromCurrentCondition(data) : undefined;
+        const errorFromPreview = data ? errorFromPreviewData(data) : undefined;
+        const error = errorFromPreview || errorFromCondition;
```

Comment:
This is the main fix - show the error that we would show upon save, inside the form